code/modules/mob/living/simple_animal/abnormality/waw/snow_whites_apple.dm

```diff
@@ -285,16 +289,14 @@ GLOBAL_LIST_EMPTY(vine_list)
 				"Even after they left, my form would not decay.","She cast me aside and left with her prince. After many days i wondered why i continued to exist.",
 				"How long has it been since the witch died... Where am i?"))
 
-/obj/structure/alien/weeds/apple_vine/proc/vinespike()
-	for(var/mob/living/L in range(0, src))
+/obj/structure/apple_vine/proc/vinespike() //called by snow white when she attacks
+	for(var/mob/living/L in get_turf(src)) //previously used inrange(0)
 		var/mob/living/carbon/human/victem = L
-		if(istype(victem, /mob/living/simple_animal/hostile/abnormality/snow_whites_apple) || ("pink_midnight" in victem.faction))
-			return
-		if(istype(victem, /mob/living/simple_animal/hostile/abnormality/scaredy_cat)) //scardy cat consideration?
-			var/mob/living/simple_animal/hostile/abnormality/scaredy_cat/catally
-			if(catally.friend == /mob/living/simple_animal/hostile/abnormality/snow_whites_apple)
-				return
-		if(victem.stat != DEAD)
+		if(victem.stat != DEAD && (!istype(victem, /mob/living/simple_animal/hostile/abnormality/snow_whites_apple) || !("pink_midnight" in victem.faction)))
+			if(istype(victem, /mob/living/simple_animal/hostile/abnormality/scaredy_cat)) //scardy cat consideration?
+				var/mob/living/simple_animal/hostile/abnormality/scaredy_cat/catally
+				if(catally.friend == /mob/living/simple_animal/hostile/abnormality/snow_whites_apple)
+					return
```

Comment:
```suggestion
		if(victim.stat == DEAD || faction_check_mob(victim))
			return
```

Reply:
Egor this is a structure. It has no faction.

Reply:
Make the proc pass snow white apple as argument so that we can check it.
I assume vines cannot strike when apple is dead, anyway, right?